### ISSUE TITLE
Check types of the string-like metadata in transactions

### ISSUE THREAD

**Issue by jodator** [Joystream/pioneer#501]
It looks like types used to provide text descriptions for some extrinsic changed to `Bytes` from previously defined.

For instance:

```
export interface BuyMembershipParameters extends Struct {
  readonly root_account: AccountId;
  readonly controller_account: AccountId;
  readonly name: Option<Text>;
  readonly handle: Option<Text>;
  readonly avatar_uri: Option<Text>;
  readonly about: Option<Text>;
  readonly referrer_id: Option<MemberId>;
}
```

changed into:

```
/** @name BuyMembershipParameters */
export interface BuyMembershipParameters extends Struct {
  readonly root_account: AccountId;
  readonly controller_account: AccountId;
  readonly handle: Option<Text>;
  readonly metadata: Bytes;
  readonly referrer_id: Option<MemberId>;
}
```

So the `name`, `about` and `avatar_uri` were migrated to `metadata` as `Bytes`.

**Comment by jodator** [comment]
I think that the best way to see how those are used is to check how the integration tests use those. Example: [https://github.com/Lezek123/substrate-runtime-joystream/blob/1bfe24ed72d3c49b3f372dd9e92b91565e5d42c1/tests/integration-tests/src/fixtures/workingGroups/ApplyOnOpeningsHappyCaseFixture.ts#L71-L85](https://github.com/Lezek123/substrate-runtime-joystream/blob/1bfe24ed72d3c49b3f372dd9e92b91565e5d42c1/tests/integration-tests/src/fixtures/workingGroups/ApplyOnOpeningsHappyCaseFixture.ts#L71-L85)

So, in other words, we have to install `@joystream/metadata-protobuf` package and use it to properly encode information.

/cc @Lezek123 Do I get the above correctly :point\_up: that we should follow the integration tests implementation in order to use properly encode any `metedata` field (assuming that this is encoded as `Bytes` in the API)?
